### PR TITLE
Clear package search cache after installing/uninstalling a package.

### LINKED ISSUE #13924: [Bug Bash] The “README” tab doesn’t show in the right panel of the PM UI after installing package
### NuGet Product Used

Visual Studio Package Management UI

### Product Version

NuGet Client Dev\6.13.0.63

### Worked before?

It’s not a regression since it is a new feature. 

### Impact

It bothers me. A fix would be nice

#### Notes:   

1. The repro rate is 70%.  

2. This issue easily reproes when installing the package for first time. 

3. Clicking the refresh button doesn’t work. 

4. It’s not a regression since it is a new feature. 

#### Repro Steps:    

1. Create a “C# Console App (.NET 9.0)” project.    

2. Right-click the project in Solution Explorer and select "Manage NuGet Packages…" menu item to open the PM UI.    

3. Click “Browse” tab and select the source “nuget.org” from Package source drop-down list. 

4. Install a package (e.g. RestSharp) and switch to “Installed" tab. 

5. Observe the details panel on the right panel of PM UI.  

#### Expected:    

The “README” tab should show in the right panel as below. 

#### Actual:    

The “README” tab doesn’t show in the right panel of the PM UI after installing package as below screenshot:  
image

### Verbose Logs

_No response_

### PR DESCRIPTION
**Pull request by jgonz120**
# Bug

Fixes: https://github.com/NuGet/Home/issues/13924
Fixes: https://github.com/NuGet/Client.Engineering/issues/3105

## Description
When a package is selected, we get its information from the search cache. However, for the README if the package was installed at some point after the cache was set, the cache results do not include the path to the README. This update allows for the cache to be reset and resets it after an install/uninstall operation has been completed. 

## PR Checklist

- [x] Meaningful title, helpful description and a linked NuGet/Home issue
- [x] Added tests
- [ ] ~Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.~

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs [hunk 1]
```diff
@@ -1307,7 +1307,7 @@ private async ValueTask RefreshAsync()
                     Model.Context.ServiceBroker,
                     Model.Context.Projects,
                     CancellationToken.None);
-                _packageList.UpdatePackageStatus(installedPackages.ToArray());
+                _packageList.UpdatePackageStatus(installedPackages.ToArray(), true);
```
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs]: Nit: use named parameter

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageSearchServiceTests.cs [hunk 2]
```diff
@@ -275,6 +275,38 @@ public async Task GetPackageVersionsAsync_WhenIsTransitiveAndCacheIsNotPopulated
...
+                {
+                    SlidingExpiration = ObjectCache.NoSlidingExpiration,
+                    AbsoluteExpiration = ObjectCache.InfiniteAbsoluteExpiration,
+                };
+
+                string cacheId = PackageSearchMetadataCacheItem.GetCacheId(packageMetadata.Identity.Id, true, packageSources);
+                var cacheEntry = new PackageSearchMetadataCacheItem(packageMetadata, metadataProvider);
+                NuGetPackageSearchService.PackageSearchMetadataMemoryCache.AddOrGetExisting(cacheId, cacheEntry, _cacheItemPolicy);
+
+                string cacheIdToRemove = PackageSearchMetadataCacheItem.GetCacheId(packageMetadataToRemove.Identity.Id, true, packageSources);
```
Comment (donnie-msft) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageSearchServiceTests.cs]: Nit: use named parameter

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageSearchServiceTests.cs [hunk 3]
```diff
@@ -275,6 +275,38 @@ public async Task GetPackageVersionsAsync_WhenIsTransitiveAndCacheIsNotPopulated
             }
         }
 
+        [Fact]
+        public void ClearFromCache_WhenCalled_PackageSearchMetadataMemoryCacheHasItemCleared()
```
Comment (donnie-msft) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageSearchServiceTests.cs]: Nit: more specific naming? Maybe just:
```suggestion
        public void ClearFromCache_WhenPackageSearchMetadataMemoryCacheHasItem_ItemCleared()
```

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs (hunk 1 above)
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs]: I recall this being a hot path for many events in PMUI. I know we're considering refactoring a lot of this code at some point which may reduce the frequency as well. 
Just wondering how confident you feel that this isn't overly clearing items from the cache? I'll look later as well.

**Review by donnie-msft — DISMISSED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs (hunk 1 above)
Comment (jgonz120) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs]: I added a bool to allow the caller of "RefreshAsync" to determine if the cache should be cleared. I'm only passing true from "ExecuteAction" which should cover install/uninstall/update.

Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs]: Awesome!

**Review by donnie-msft — APPROVED** [review]

**Review by jeffkl — APPROVED** [review]

**Pull request merged**